include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -82,6 +82,196 @@ namespace internal
+              for (unsigned int i = 0; i < fe->size(); ++i)
+                switch ((*fe)[i].reference_cell_type())
+                  {
+                    case ReferenceCell::Type::Vertex:
+                    case ReferenceCell::Type::Line:
+                    case ReferenceCell::Type::Quad:
+                    case ReferenceCell::Type::Hex:
+                      needs_hypercube_setup |= true;
+                      break;
+                    case ReferenceCell::Type::Tri:
+                    case ReferenceCell::Type::Tet:
+                      needs_simplex_setup |= true;
+                      break;
+                    case ReferenceCell::Type::Wedge:
+                      needs_wedge_setup |= true;
+                      break;
+                    case ReferenceCell::Type::Pyramid:
+                      needs_pyramid_setup |= true;
+                      break;
+                    default:
+                      Assert(false, ExcNotImplemented());
+                  }
+
+            std::unique_ptr<dealii::Quadrature<dim>> quadrature_simplex;
+            std::unique_ptr<dealii::Quadrature<dim>> quadrature_hypercube;
+            std::unique_ptr<dealii::Quadrature<dim>> quadrature_wedge;
+            std::unique_ptr<dealii::Quadrature<dim>> quadrature_pyramid;
+
+            if (needs_simplex_setup)
+              {
+                Assert(1 <= n_subdivisions && n_subdivisions <= 2,
+                       ExcNotImplemented());
+                quadrature_simplex = std::make_unique<Quadrature<dim>>(
+                  Simplex::FE_P<dim, spacedim>(n_subdivisions == 1 ? 1 : 2)
+                    .get_unit_support_points());
+              }
+
+            if (needs_hypercube_setup)
+              {
+                quadrature_hypercube =
+                  std::make_unique<QIterated<dim>>(QTrapezoid<1>(),
+                                                   n_subdivisions);
+              }
+
+            if (needs_wedge_setup)
+              {
+                Assert(1 <= n_subdivisions && n_subdivisions <= 2,
+                       ExcNotImplemented());
+                quadrature_wedge = std::make_unique<Quadrature<dim>>(
+                  Simplex::FE_WedgeP<dim, spacedim>(
+                    1 /*note: vtk only supports linear wedges*/)
+                    .get_unit_support_points());
+              }
+
+            if (needs_pyramid_setup)
+              {
+                Assert(1 <= n_subdivisions && n_subdivisions <= 2,
+                       ExcNotImplemented());
+
+                std::vector<Point<dim>> points;
+
+                points.emplace_back(-1.0, -1.0, 0.0);
+                points.emplace_back(+1.0, -1.0, 0.0);
+                points.emplace_back(+1.0, +1.0, 0.0);
+                points.emplace_back(-1.0, +1.0, 0.0);
+                points.emplace_back(+0.0, +0.0, 1.0);
+
+                quadrature_pyramid = std::make_unique<Quadrature<dim>>(points);
+              }
```

Comment:
I am not sure if this is the correct place. But I need to enumerate the vertices for VTK differently.